File: android/dito-sdk/src/main/java/br/com/dito/ditosdk/notification/DitoNotificationOptions.kt
package br.com.dito.ditosdk.notification

import androidx.annotation.ColorInt
import androidx.annotation.DrawableRes

data class DitoNotificationOptions(
    @DrawableRes val smallIconResId: Int? = null,
    @DrawableRes val largeIconResId: Int? = null,
    val soundResourceName: String? = null,
    @ColorInt val accentColor: Int? = null,
    val badgeEnabled: Boolean = true
)

File: flutter/lib/dito_notification_options.dart
class DitoNotificationOptions {
  final int? accentColor;
  final int? largeIconResId;
  final int? smallIconResId;
  final String? soundResourceName;

  const DitoNotificationOptions({
    this.accentColor,
    this.largeIconResId,
    this.smallIconResId,
    this.soundResourceName,
  });

  Map<String, dynamic> toMap() {
    return {
      'accentColor': accentColor,
      'largeIconResId': largeIconResId,
      'smallIconResId': smallIconResId,
      'soundResourceName': soundResourceName,
    };
  }
}
